### PR TITLE
added an archives page for marathon

### LINKED ISSUE #544: Create an archives page for marathons + reduce featured number of marathons on home page
One of the biggest barriers to accepting/creating more marathons is the amount of screen space it takes on the home page. We'll need to create another archives page for it.

### PR DESCRIPTION
**Pull request by mliu59**
- added an archives page for marathons, nearly a carbon copy of the sprints archive. It also uses the limit/offset pagination. Default limit of prompts on a page of the archive is 10 (as opposed to the 20 for sprints)
- reduced featured list size on home page to 5. It will only show the most recent 5 (sorted by prompt id)
image
- marathon archive will also check for saved games. If there is a saved game for a prompt, the play link will be "continue" to continue saved game instead. 
image
- Mobile looks the same as the home menu table.

### CODE REVIEW

File: wikispeedruns/prompts.py [hunk 1]
```diff
@@ -249,12 +256,17 @@ def get_archive_prompts(prompt_type: PromptType, user_id: Optional[int]=None, of
     '''
     Get all prompts for archive, including currently active
     '''
-    if (prompt_type == "sprint"):
-        query = "SELECT prompt_id, start, end, rated, active_start, active_end, cmty_added_by, cmty_anonymous, cmty_submitted_time, username FROM sprint_prompts"
+    #if (prompt_type == "sprint"):
+    #    query = "SELECT prompt_id, start, end, rated, active_start, active_end, cmty_added_by, cmty_anonymous, cmty_submitted_time, username FROM sprint_prompts"
     # elif (prompt_type == "marathon")
```
Comment (dqian3) [inline wikispeedruns/prompts.py:261]: Oh wow, I guess we were thinking about this a while ago/ Also maybe just get rid of these comments.

File: apis/marathon_api.py [hunk 2]
```diff
@@ -122,6 +123,27 @@ def get_all_marathon_prompts():
...
+    ORDER BY prompt_id DESC
+    LIMIT %(p_limit)s
+    """
+
+    db = get_db()
+    with db.cursor(cursor=DictCursor) as cursor:
+        cursor.execute(query, args)
+        results = cursor.fetchall()
+
+        return jsonify(results)
```
Comment (dqian3) [inline apis/marathon_api.py:146]: Is this just a special case of the archive endpoint? If so you might just want to use  the `prompts.get_archive_prompts` to reduce code duplication? Or are there fields that aren't there?

**Review by dqian3 — APPROVED** [review]
Looks good, and seems like you tested it. A few small comments

File: apis/marathon_api.py (hunk 2 above)
Comment (mliu59) [reply apis/marathon_api.py]: Fixed. This was duplicate code

File: wikispeedruns/prompts.py (hunk 1 above)
Comment (mliu59) [reply wikispeedruns/prompts.py]: Removed

**Pull request merged**